### PR TITLE
31 reusable table component

### PR DESCRIPTION
**Pull request by tehreemAsif63**
## Description

This feature introduces a reusable table component that displays patient and doctor information in separate contexts. For doctors, the patient table includes first names, last names, and personal numbers, facilitating quick access to health data received from the Moberg monitor. For admins, the doctor table shows first names, last names, personal numbers (replaced with SSL), usernames, and passwords, enabling effective management of doctor accounts.

- **Related Issue:** #31  
- **Feature Name:** Reusable Table Component for Patients and Doctors

## Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (non-breaking improvement to an existing feature)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

Please describe the tests that you ran to verify the feature. Provide instructions so others can replicate these tests. Include any relevant details like the test environment, test cases, and scripts.

- **Tested On:**
  - [x] Local machine
  - [ ] CI Pipeline
  - [ ] Staging environment
- **Test Description:**
  - [ ] Unit tests
  - [ ] Integration tests
  - [x] Manual testing

## Checklist

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code where necessary.
- [x] I have added documentation explaining what this feature does and how to use it.
- [x] I have added tests to cover my changes.

- [ ] I have checked for any security concerns or performance impacts.

## Impacted Areas in the Project

Please specify which areas of the project this feature might affect.

- **Components/Files Affected:** Table, DoctorTable, PatientTable, AdminPage
- **Other features or services impacted:** User authentication and authorization checks for data management.

## Additional Information

This feature relies on a well-defined database schema and the necessary API endpoints for dynamic data loading. Implementing user authentication ensures only authorized users can edit or delete entries, enhancing security and functionality.

---

### Reviewer’s Guide:
Please ensure:
- Code is clean and properly documented.
- The feature works as expected and doesn’t introduce any regressions.

### CODE REVIEW

File: client/src/views/AdminsPage.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import DoctorTable, {Doctor} from '../components/DoctorTable';
+import { useNavigate } from 'react-router-dom'; // Import useNavigate
+
+
+const AdminPage: React.FC = () => {
+    const navigate = useNavigate(); // Initialize useNavigate
+    const [doctorData, setDoctorData] = useState<Doctor[]>([
+      { _id: '111', firstName: 'John', lastName: 'Doe', SSN: '987654', username: 'jdoe', password: 'password123' },
+      { _id: '222', firstName: 'Jane', lastName: 'Smith', SSN: '456789', username: 'jsmith', password: 'password456' }
+    ]);
```
Comment (Flam30) [inline client/src/views/AdminsPage.tsx:11]: Has to be altered after integration to use json array, not hardcoded data.

File: client/src/views/DoctorsPage.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+const PatientListPage: React.FC = () => {
+  const patientData: Patient[]  = [ 
+    {'_id': '123', firstName: 'David', lastName: 'Svenson', SSN: '200205156835' },
+    {'_id': '321', firstName: 'John', lastName: 'Doe', SSN: '198504121234' },
+    {'_id': '567', firstName: 'Jane', lastName: 'Smith', SSN: '199203058765' },
+    {'_id': '234', firstName: 'Emily', lastName: 'Johnson', SSN: '197812253456' },
+    {'_id': '568', firstName: 'Michael', lastName: 'Brown', SSN: '200001019876' },
+    {'_id': '891', firstName: 'Sarah', lastName: 'Davis', SSN: '199507172345' }
+  ];
```
Comment (Flam30) [inline client/src/views/DoctorsPage.tsx:13]: Re JSON array comment.

**Review by Flam30 — APPROVED** [review]
Good additions. There are edits which will need to be made before the feature is complete, however this PR achieves 80% of what needs to be done.

Approved for merging.

File: client/src/components/Table.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+
+// Define types for the props
+interface TableProps {
+  columns: string[];
+  data: Array<{ [key: string]: any }>;  // Array of objects representing the table rows, each with key-value pairs for cell data
+  onEdit?: (row: any) => void; // Optional edit function for admin
+  onDelete?: (row: any) => void; // Optional delete function for admin
```
[the remark this answers is not in the document]
Comment (tehreemAsif63) [reply client/src/components/Table.tsx]: No, they don't have to be. Thank you for pointing that out! I've updated the type definition to ensure type safety. I hope it's alright now. What do you think?

File: client/src/views/AdminsPage.tsx (hunk 1 above)
Comment (tehreemAsif63) [reply client/src/views/AdminsPage.tsx:11]: Absolutely, once we integrate with the endpoints for doctors and patients, we can fetch real data and replace the dummy data placeholders. This should be straightforward to implement.

File: client/src/views/DoctorsPage.tsx (hunk 2 above)
Comment (tehreemAsif63) [reply client/src/views/DoctorsPage.tsx:13]: Yes, after integration we can fetch real data and replace the dummy data placeholders.

File: client/src/components/Table.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (tehreemAsif63) [reply client/src/components/Table.tsx:8]: I made the necessary changes, but the pipeline wasn't passing due to other issues. Once we input the data, we can adjust and fix the types accordingly to ensure type consistency every where.

**Pull request merged**